### PR TITLE
Improve accessibility

### PR DESCRIPTION
**Pull request by dge808**
### Description

Fixes a bug where geosuggest suggestions were not read aloud by the screen readers 

### Checklist

- [x] All tests passing
- [x] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary
- [x] Commits and PR follow conventions

### CODE REVIEW

File: src/Geosuggest.tsx [hunk 1]
```diff
@@ -100,6 +105,9 @@ export default class extends React.Component<IProps, IState> {
     this.onSuggestNoResults = this.onSuggestNoResults.bind(this);
     this.hideSuggests = this.hideSuggests.bind(this);
     this.selectSuggest = this.selectSuggest.bind(this);
+    this.listId = `geosuggest__list_${Math.random()
```
Comment (KittyGiraudel) [inline src/Geosuggest.tsx:108]: Shall we make it possible to pass a list ID, and default to this value if none is passed?

File: src/Geosuggest.tsx [hunk 2]
```diff
@@ -594,6 +602,9 @@ export default class extends React.Component<IProps, IState> {
         onPrev={this.onPrev}
         onSelect={this.onSelect}
         onEscape={this.hideSuggests}
+        isSuggestHidden={this.state.isSuggestsHidden}
```
Comment (KittyGiraudel) [inline src/Geosuggest.tsx:605]: The state key is called `isSuggestsHidden`, and you pass down `isSuggestHidden`. Maybe we keep the same for consistency?

File: src/Geosuggest.tsx [hunk 3]
```diff
@@ -615,6 +626,7 @@ export default class extends React.Component<IProps, IState> {
         onSuggestMouseOut={this.onSuggestMouseOut}
         onSuggestSelect={this.selectSuggest}
         renderSuggestItem={this.props.renderSuggestItem}
+        listId={this.listId}
```
Comment (KittyGiraudel) [inline src/Geosuggest.tsx:629]: Shall we pass `id` down instead of `listId` since it’s the `SuggestList` component?

File: src/suggest-item.tsx [hunk 4]
```diff
@@ -161,7 +161,10 @@ export default class extends React.PureComponent<IProps, {}> {
         style={this.props.style}
         onMouseDown={this.props.onMouseDown}
         onMouseOut={this.props.onMouseOut}
-        onClick={this.onClick}>
+        onClick={this.onClick}
+        role="option"
+        aria-selected={this.props.isActive}
+        id={suggest.placeId}>
```
Comment (KittyGiraudel) [inline src/suggest-item.tsx:167]: Can this safely be used as a DOM id?

File: src/Geosuggest.tsx (hunk 1 above)
Comment (dge808) [reply src/Geosuggest.tsx:108]: This functionality was not provided by geosuggest before, `<ul>`, there were not an id's on UL before, so I decided to live it as it is, without adding new functionality

File: src/Geosuggest.tsx (hunk 2 above)
Comment (dge808) [reply src/Geosuggest.tsx:605]: sure, it's a typo

File: src/Geosuggest.tsx (hunk 3 above)
Comment (dge808) [reply src/Geosuggest.tsx:629]: we could 🙂

**Pull request closed without merging**